File: Add Code Here/LeetCode_Sols/Medium/974. Subarray Sums Divisible by K/Solution.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> map;
        int sum = 0;
        int count = 0;
        for(int i = 0; i < n; i++) {
            sum += nums[i];
            if(sum % k == 0) count++;
            int rem = sum % k;
            if(rem < 0) rem += k;
            if(map[rem]) count += map[rem];
            map[rem] = map[rem] + 1;
        }
        return count;
    }
};

File: Add Code Here/LeetCode_Sols/Medium/974. Subarray Sums Divisible by K/Solution.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        n = len(nums)
        map = {}
        sum = 0
        count = 0
        for i in range(n):
            sum = sum + nums[i]
            if sum % k == 0: count = count + 1
            rem = sum % k
            if rem < 0: rem = rem + k
            if map.get(rem):
                count = count + map[rem]
                map[rem] = map[rem] + 1
            else:
                map[rem] = 1
        return count
